server/plugin.go

```diff
@@ -0,0 +1,139 @@
+	// loads the details back so everything works
+	p.LoadMeetingsFromStore()
+
+	if err := p.OnConfigurationChange(); err != nil {
+		return err
+	}
+	config := p.config()
+	if err := config.IsValid(); err != nil {
+		return err
+	}
+
+	bbbAPI.SetAPI(config.BASE_URL+"/", config.SALT)
+
+	//every 2 minutes, look through active meetings and check if recordings are done
+	p.c = cron.New()
+	p.c.AddFunc("@every 2m", p.Loopthroughrecordings)
+	p.c.Start()
+
+	// register slash command '/bbb' to create a meeting
+	return p.API.RegisterCommand(&model.Command{
+		Trigger:          "bbb",
+		AutoComplete:     true,
+		AutoCompleteDesc: "Create a BigBlueButton meeting",
+	})
+}
+
+
+//following method is to create a meeting from '/bbb' slash command
+func (p *Plugin) ExecuteCommand(c *plugin.Context, args *model.CommandArgs) (*model.CommandResponse, *model.AppError) {
+
+	meetingpointer := new(dataStructs.MeetingRoom)
+	err := p.PopulateMeeting(meetingpointer, nil, "") 
+	if err != nil {
+		return nil, model.NewAppError("ExecuteCommand", "Please provide a 'Site URL' in Settings > General > Configuration", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	p.createStartMeetingPost(args.UserId, args.ChannelId, meetingpointer)
+	p.Meetings = append(p.Meetings, *meetingpointer)
+	return &model.CommandResponse{}, nil
+
+}
+
+//this is the router to handle our server calls
+//methods are all in responsehandlers.go
+func (p *Plugin) ServeHTTP(c *plugin.Context,w http.ResponseWriter, r *http.Request) {
+
+	config := p.config()
+	if err := config.IsValid(); err != nil {
+		http.Error(w, "This plugin is not configured.", http.StatusNotImplemented)
+		return
+	}
+
+	path := r.URL.Path
+	if path == "/joinmeeting" {
+		p.handleJoinMeeting(w, r)
+	} else if strings.HasPrefix(path, "/endmeeting") {
+		p.handleEndMeeting(w, r)
+	} else if path == "/create" {
+		p.handleCreateMeeting(w, r)
+	} else if strings.HasPrefix(path, "/recordingready") {
+		p.handleRecordingReady(w, r)
+	} else if path == "/getattendees" {
+		p.handleGetAttendeesInfo(w, r)
+	} else if path == "/publishrecordings" {
+		p.handlePublishRecordings(w, r)
+	} else if path == "/deleterecordings" {
+		p.handleDeleteRecordings(w, r)
+	} else if strings.HasPrefix(path, "/meetingendedcallback") {
+		p.handleImmediateEndMeetingCallback(w, r, path)
+	} else if path == "/ismeetingrunning" {
+		p.handleIsMeetingRunning(w, r)
+	} else if path == "/redirect"{
+		// html file to automatically close a window
+			fmt.Fprintf(w,`<!doctype html><html><head><script>
+				 								window.onload = function load() {
+													window.open('', '_self', '');
+													window.close();
+													};
+											</script></head><body></body></html>`)
+	}else {
+		http.NotFound(w, r)
+	}
+	return
+}
+
+func (p *Plugin) OnDeactivate() error {
+	//on deactivate, save meetings details, stop check recordings looper, destroy webhook
+	p.SaveMeetingToStore()
```

Comment:
This is not save. A plugin can crash at any time and the data will be lost. You should write to the store on any changes.

Reply:
Will be handled in https://github.com/blindsidenetworks/mattermost-plugin-bigbluebutton/issues/61